### PR TITLE
dashlane-cli 1.1.0 (new formula)

### PR DESCRIPTION
**Pull request by Mikescops**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Publishing `dashlane-cli` (https://github.com/Dashlane/dashlane-cli) to homebrew. Support for arm and x86/64 has been tested.

### CODE REVIEW

File: Formula/dashlane-cli.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    arch = `uname -m`
+    system "npm", "ci"
+    system "npm", "run", "build"
+    system "./prepare-pkg.sh", libc
+    system "npx", "pkg", ".", "-t", "node16-#{platform}-#{arch.chomp}", "-o", "dcli"
+    bin.install "dcli"
+  end
+
+  test do
+    assert_equal version.to_s, shell_output("#{bin}/dcli --version").chomp
```
Comment (ZhongRuoyu) [inline Formula/dashlane-cli.rb:35]: Can you add a check that at least covers some of its functionality? Version checks like this almost never fail.

File: Formula/dashlane-cli.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  sha256 "6345319412b5d224083bc032f4a1ba1890ace5ecf70a63f4e9930aa6537d1d49"
+  license "Apache-2.0"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+    regex(%r{href=.*?/tag/vv?(\d+(?:\.\d+)+)["' >]}i)
+  end
+
+  depends_on "node@16"
```
[the remark this answers is not in the document]
Comment (Mikescops) [reply Formula/dashlane-cli.rb:16]: we can't use node@18 as our tool is not yet ready for it, we will update in time before 16 is no more supported

File: Formula/dashlane-cli.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  on_linux do
+    depends_on "glibc"
+  end
```
[the remark this answers is not in the document]
Comment (Mikescops) [reply Formula/dashlane-cli.rb:20]: good to know thanks

File: Formula/dashlane-cli.rb (hunk 1 above)
Comment (Mikescops) [reply Formula/dashlane-cli.rb:35]: All other commands requires inputs from users with an active account on Dashlane service, if you know a way to input secrets in the tests I'll be happy to add extra tests.

File: Formula/dashlane-cli.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  desc "Command-line interface for Dashlane"
+  homepage "https://dashlane.com"
+  url "https://github.com/Dashlane/dashlane-cli/archive/refs/tags/v1.1.0.tar.gz"
+  sha256 "6345319412b5d224083bc032f4a1ba1890ace5ecf70a63f4e9930aa6537d1d49"
+  license "Apache-2.0"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+    regex(%r{href=.*?/tag/vv?(\d+(?:\.\d+)+)["' >]}i)
```
[the remark this answers is not in the document]
Comment (Mikescops) [reply Formula/dashlane-cli.rb:13]: ok noted

File: Formula/dashlane-cli.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/dashlane-cli.rb:30]: How about checking if it fails correctly?

Comment (ZhongRuoyu) [reply Formula/dashlane-cli.rb:30]: > if you know a way to input secrets

That's not possible, unfortunately.

**Comment by ZhongRuoyu** [comment]
Thanks for your input, @Mikescops.

I had a look at the https://github.com/Dashlane/dashlane-cli repo, but I noted that this project does not seem to be notable for inclusion. In fact, if you run `brew audit --new dashlane-cli` (as suggested in the PR template), you'll see this:

```
$ brew audit --new dashlane-cli
dashlane-cli:
  * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
Error: 1 problem in 1 formula detected
```

Please rest assured that this has nothing to do with the quality of the software. It's just that there are already thousands of packages at homebrew/core, and we can't unconditionally accept more, so we have to draw a line somewhere, sorry.

You may want to consider hosting this formula in a [tap](https://docs.brew.sh/Taps). In fact, some taps are already [used by many](https://docs.brew.sh/Interesting-Taps-and-Forks). To get started, please read https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap.

Nevertheless, thanks for submitting the PR!

**Pull request closed without merging**